pkg/csi/cinder/controllerserver.go

```diff
@@ -398,6 +402,23 @@ func (cs *controllerServer) ControllerGetCapabilities(ctx context.Context, req *
 }
 
 func (cs *controllerServer) ValidateVolumeCapabilities(ctx context.Context, req *csi.ValidateVolumeCapabilitiesRequest) (*csi.ValidateVolumeCapabilitiesResponse, error) {
+	volumeId := req.GetVolumeId()
+	if len(volumeId) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "ValidateVolumeCapabilities request missing Volume Name")
+	}
+
+	if req.VolumeCapabilities == nil {
+		return nil, status.Error(codes.InvalidArgument, "ValidateVolumeCapabilities request missing capabilities")
+	}
+
+	_, err := cs.Cloud.GetVolume(volumeId)
+	if err != nil {
+		if cpoerrors.IsNotFound(err) {
+			return nil, status.Error(codes.NotFound, "ValidateVolumeCapabilities Volume not found")
+		}
+		return nil, status.Error(codes.Internal, fmt.Sprintf("ValidateVolumeCapabilities failed with error %v", err))
+	}
+
 	return nil, status.Error(codes.Unimplemented, fmt.Sprintf("ValidateVolumeCapabilities is not yet implemented"))
```

Comment:
@jichenjc Thanks for your PR but it is good if we could implement volume capabilties with it

Reply:
yes, I am thinking how to add this ... but not bad to add this as it can at least report some error  and fix sanity check as well.